### ISSUE TITLE
Feature Request: Better Activity Scheduling for Handlers

### ISSUE THREAD

**Issue by Cyb3r-Monk** [cmu-sei/GHOSTS#195]
When using several handlers, every activity can happen simultaneously/concurrently. For example, the agent may browse the internet, create a word document, send an email, all at the same time. How can we avoid this? 

Here is what I want to do for an arbitrary handler:
```
Browser
    Every hour:
        browse the internet only for 15 minutes and stop(but don't close the browser)

Word
   Every 30 minutes:
       Create/Modify a document
```

Another approach for better activity simulation could be:
1. Create activity classes(Browsing, Word, Excel, outlook, etc.)
2. Assign activities to the agent
3. Agent picks an activity(predefined order or random selection based on weight or NPC profile(more Word, less Excel, etc.))
4. Agent performs the selected activity. When performing the activity it doesn't perform other activities and wait for the activity completion(`delayafter` can be used here)

This way, the agent first checks emails and replies an email. Then, it performs some browsing for a while. Then, it can create/modify an Office document. 

As a configuration, I can define working hours and allocate time duration to the activities. Then, the agent starts performing activities in an order or randomly. At the end of the working hours, the agent stops whatever it's doing and goes home, spends time with the family 😄

**Comment by sei-dupdyke** [comment]
Currently, you could have the default timeline for an agent effectively have them do nothing — no handlers at all — and then script simple posts to the API to send particular activity timelines down to clients at specific times. I can help you with this, because I probably need to document it far better than currently exists.

For some backstory, these specific timing use cases seem very different from our original intent of — generally speaking — agents would be "set and forget" with an appropriate amount of "realistic" randomization. Specific events would be related to injects or events we're managing in particular for a training or exercise.

I'm not opposed to supporting these kinds of requests at the client level, but I want to be careful not to force people to modify configurations that are already out there performing under the "sufficiently random" model. 

Which, I suppose is the problem every product faces upon some level of maturation.

TLDR; let me think about how best to implement this beside current functionality. I'm afraid that makes it not a fast turn around, and maybe something that has to wait for a major release, sorry.

**Comment by Cyb3r-Monk** [comment]
I see. My proposal is still being realistic, but improving the realisticity(is there such a word?). To give you an example, I just don't want the NPC to create a Word document and browse the internet at the same time. Or, I wouldn't prefer the agent perform the same activity at the same period like every hour(even with some jitter involved). 

The "order" in my first comment wasn't meant to be defining activities to be performed in a specific order which would kill the randomness. In the injection scenarios, maybe an activity picker class first checks if there is something coming from the API server, then runs it (if there is) or continues to randomly picking up activities and performing them. The agent may need multitasking, though(browse the internet while the Word document is open).

Currently, when the agent creates a document, it creates and saves the document in just a few seconds. The browsing happens indefinitely without giving a break.  In my opinion, certain activities should have a duration when they are kicked off. How about something like below?
Browsing:  "perform browsing for 20 minutes"
Wrod/Excel/PowerPoint: "create/open existing document, write something into it, wait for 10 minutes, save and close" (more realistic document handling) 

So, for the Browser handler:
A new parameter `duration` which depends on the cron scheduler. Cron scheduler can enable the looping when the schedule kicks in, then disables it when the time is up according to the duration. 

For the document handlers:
A new parameter `duration` that defines how long the application should stay open before saving the document and closing it. Jitter can be applied to the duration, too(would be great!)

Browser handler improvement looks tricky, but I think the document handler improvement would be a great addition.
